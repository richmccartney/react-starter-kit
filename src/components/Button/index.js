// @flow
/** @jsx jsx */

import * as React from 'react'
import { css, jsx } from '@emotion/core'

type PropType = {
  children: React.Node,
  name?: string,
  href?: string,
  disabled?: boolean,
  onClick: Function,
  ariaLabel?: string
}

const buttonStyle = css`
  appearance: none;
  background-color: #b0d7ff;
  border: 1px solid rgba(0, 0, 0, 0.1);
  border-radius: 4px;
  box-sizing: border-box;
  color: #2d3142;
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  padding: 8px 12px 6px;
  text-align: center;
  text-decoration: none;
  user-select: none;
`

class Button extends React.Component<PropType> {
  static defaultProps = {
    disabled: false,
    onClick: () => {}
  }

  onClick = (e: Event) => {
    const { disabled } = this.props
    if (disabled) {
      return
    }
  }

  render = () => {
    const { children, name, disabled, onClick, href, ariaLabel } = this.props

    const Tag = href ? 'a' : 'button'

    return (
      <Tag
        css={css`
          ${buttonStyle}
        `}
        href={Tag === 'a' ? href : undefined}
        onClick={onClick}
        disabled={disabled}
        name={name}
        aria-label={ariaLabel}
      >
        {children}
      </Tag>
    )
  }
}

export default Button
