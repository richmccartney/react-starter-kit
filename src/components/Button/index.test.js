import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Button from './'

describe('Button', () => {
  it('should render a default Button', () => {
    const button = renderer.create(
      <div>
        <Button>Button</Button>
      </div>
    )
    const json = button.toJSON()
    expect(json).toMatchSnapshot()
  })

  it('should render an anchor element if href exsists', () => {
    const button = renderer.create(
      <div>
        <Button href="link">Button</Button>
      </div>
    )
    const json = button.toJSON()
    expect(json).toMatchSnapshot()
  })

  it('should render an button with an aria label', () => {
    const button = shallow(<Button ariaLabel="Test">Link</Button>)
    expect(button.find('button').prop('aria-label')).toEqual('Test')
  })

  it('onClick is called', () => {
    const onClick = jest.fn()
    const button = shallow(<Button onClick={onClick}>Ok!</Button>)

    button.find('button').simulate('click')
    expect(onClick.mock.calls.length).toEqual(1)
  })

  it('is not called when is disabled', () => {
    const onClick = jest.fn()
    const button = shallow(
      <Button disabled onClick={onClick}>
        onClick test
      </Button>
    )

    button.find('button').simulate('click')
    expect(onClick.length).toEqual(0)
  })
})
