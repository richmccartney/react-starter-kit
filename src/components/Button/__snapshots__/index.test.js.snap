// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a default Button 1`] = `
<div>
  <button
    className="css-ewlrv4"
    disabled={false}
    onClick={[Function]}
  >
    Button
  </button>
</div>
`;

exports[`Button should render an anchor element if href exsists 1`] = `
<div>
  <a
    className="css-ewlrv4"
    disabled={false}
    href="link"
    onClick={[Function]}
  >
    Button
  </a>
</div>
`;
